### ISSUE TITLE
Add Async Support

### ISSUE THREAD

**Issue by Archmonger** [jazzband/django-downloadview#193]
Currently, the DownloadView operates as a sync view. It's very possible (and much more efficient) to do everything async.

File reading will have to be done via `aiofile` in order to not break the ASGI event queue though. After this change, Django versions will need to be limited to 3.1+

**Comment by Natim** [comment]
Would you be interested to file a PR implementing this?

**Comment by Archmonger** [comment]
Yes, but my time is spread a bit thin across the projects I currently support. It might take a while for me to get to this.

**Comment by Archmonger** [comment]
Note: I have an [in-progress PR](https://github.com/evansd/whitenoise/pull/359) which has an implemenation of [sending files via async Django](https://github.com/Archmonger/whitenoise/blob/0e42cb8100dd0e2fe72ed6ffabda468d26eead4e/src/whitenoise/asgi.py#L71-L82). The implementation in this repo would be heavily similar.

I believe Django supports async iterators within `StreamingHttpResponse` now, so this should be pretty simple to implement.

**Comment by Archmonger** [comment]
I still don't have time to do this PR, but I'm revisiting this issue to add some more details.

I did [some testing](https://github.com/mosquito/aiofile/issues/88#issuecomment-2314380621) a while back and determined that, within the current state of Python, the best way of performing file reading is via python stdlib threads.

I ultimately forked `whitenoise` to develop `ServeStatic`. Within my new repo, there are some examples of performant [Django ASGI file responses](https://github.com/Archmonger/ServeStatic/blob/a270833b978ff39f19817f58cda9cbe4bcbde0ad/src/servestatic/middleware.py#L128-L131) which ultimately result in [async reading](https://github.com/Archmonger/ServeStatic/blob/a270833b978ff39f19817f58cda9cbe4bcbde0ad/src/servestatic/utils.py#L102-L168).
